src/components/hoc/withWallet.tsx

```diff
@@ -25,10 +26,15 @@ export function withWallet<Props extends { wallet: EdgeCurrencyWallet }>(
     const wallet = account.currencyWallets[route.params.walletId]
 
     React.useEffect(() => {
-      if (wallet == null) navigation.goBack()
+      if (wallet == null) navigation.navigate('loanDashboard', {})
```

Comment:
This isn't a good spot to add loan-specific navigation since this HOC could be used anywhere. Also, this component isn't used in any loans scenes I suspect you meant to edit withLoanAccount instead?